api/lanes.py

```diff
@@ -1,91 +1,140 @@
-        languages = language_list(language_set)
-        seen_languages = seen_languages.union(set(languages))
-        top_level_lanes.extend(lanes_for_large_collection(_db, library, language_set))
+    # Expunge the Lanes from the database before they go into the WorkList,
+    # since it will be used across sessions.
+    map(_db.expunge, top_level_lanes)
 
-    for language_set in Configuration.small_collection_languages(library):
-        languages = language_list(language_set)
-        seen_languages = seen_languages.union(set(languages))
-        top_level_lanes.append(lane_for_small_collection(_db, library, language_set))
+    if len(top_level_lanes) == 1:
+        return top_level_lanes[0]
 
-    other_languages_lane = lane_for_other_languages(_db, library, seen_languages)
-    if other_languages_lane:
-        top_level_lanes.append(other_languages_lane)
+    wl = WorkList()
+    wl.initialize(
+        library, display_name=_("Collection"), children=top_level_lanes
+    )
+    return wl
+
+def create_default_lanes(_db, library):
+    """Reset the lanes for the given library to the default.
+
+    The database will have the following top-level lanes for
+    each large-collection:
+    'Adult Fiction', 'Adult Nonfiction', 'Young Adult Fiction',
+    'Young Adult Nonfiction', and 'Children'.
+    Each lane contains additional sublanes.
+    If an NYT integration is configured, there will also be a
+    'Best Sellers' top-level lane.
+
+    The database will also have a top-level lane named after each
+    small-collection language. Each such sublane contains "Adult
+    Fiction", "Adult Nonfiction", and "Children/YA" sublanes.
+
+    Finally the database includes an "Other Languages" top-level lane
+    which covers all other languages known to the collection.
+
+    If run on a Library that already has Lane configuration, this can
+    be an extremely destructive method. All new Lanes will be visible
+    and all Lanes based on CustomLists (but not the CustomLists
+    themselves) will be destroyed.
+    """
+    # Delete existing lanes.
+    for lane in _db.query(Lane).filter(Lane.library_id==library.id):
+        _db.delete(lane)
+
+    top_level_lanes = []
+
+    large = Configuration.large_collection_languages(library) or []
+    small = Configuration.small_collection_languages(library) or []
+    tiny = Configuration.tiny_collection_languages(library) or []
+
+    # If there are no language configuration settings, estimate the
+    # current collection size to determine the lanes.
+    if not large and not small and not tiny:
+        estimates = library.estimated_holdings_by_language()
+        [(ignore, largest)] = estimates.most_common(1)
+        for language, count in estimates.most_common():
+            if count > largest * 0.1:
+                large.append(language)
+            elif count > largest * 0.01:
+                small.append(language)
+            else:
+                tiny.append(language)
```

Comment:
These cutoffs are pretty arbitrary.

Reply:
I think I would prefer cutoffs based on absolute numbers but I also like always treating the largest collection as a 'large' and once you do that you can't do the cutoffs with absolute numbers. So this is good for now.